Corona-Warn-App/src/main/java/de/rki/coronawarnapp/eventregistration/storage/retention/EventRegistrationDbCleanUpPeriodicWorker.kt

```diff
@@ -0,0 +1,34 @@
+package de.rki.coronawarnapp.eventregistration.storage.retention
+
+import android.content.Context
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedFactory
+import dagger.assisted.AssistedInject
+import de.rki.coronawarnapp.util.worker.InjectedWorkerFactory
+import timber.log.Timber
+
+class EventRegistrationDbCleanUpPeriodicWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted private val workerParams: WorkerParameters,
+    private val traceLocationCleaner: TraceLocationCleaner,
+    private val checkInCleaner: CheckInCleaner
+) : CoroutineWorker(context, workerParams) {
+
+    override suspend fun doWork(): Result {
+        Timber.d("Starting work in EventRegistrationDbCleanUpPeriodicWorker")
+        try {
+            traceLocationCleaner.cleanUp()
```

Comment:
How about foreign key relations? Are they respected?

Reply:
There are no foreign key relations between `TraceLocations` and `CheckIns`.